src/webgpu/api/operation/resource_init/buffer.spec.ts

```diff
@@ -212,3 +215,106 @@ array buffer of getMappedRange() and the GPUBuffer itself have all been initiali
 
     await t.CheckGPUBufferContent(buffer, bufferUsage, expectedData);
   });
+
+g.test('copy_buffer_to_buffer_copy_source')
+  .desc(
+    `Verify when the first usage of a GPUBuffer is being used as the source buffer of
+CopyBufferToBuffer(), the contents of the GPUBuffer have already been initialized to 0.`
+  )
+  .fn(async t => {
+    const bufferSize = 32;
+    const bufferUsage = GPUBufferUsage.COPY_SRC;
+    const buffer = t.device.createBuffer({
+      size: bufferSize,
+      usage: bufferUsage,
+    });
+
+    const expectedData = new Uint8Array(bufferSize);
+    await t.CheckGPUBufferContent(buffer, bufferUsage, expectedData);
```

Comment:
nit: add a small comment explaining the copyBufferToBuffer happens inside that call. Makes the test depend on that not changing, which is probably fine?